symfony/framework-bundle/4.2/config/bootstrap.php

```diff
@@ -4,16 +4,21 @@
 
 require dirname(__DIR__).'/vendor/autoload.php';
 
-if (!array_key_exists('APP_ENV', $_SERVER)) {
-    $_SERVER['APP_ENV'] = $_ENV['APP_ENV'] ?? null;
-}
-
-if ('prod' !== $_SERVER['APP_ENV']) {
-    if (!class_exists(Dotenv::class)) {
-        throw new RuntimeException('The "APP_ENV" environment variable is not set to "prod". Please run "composer require symfony/dotenv" to load the ".env" files configuring the application.');
-    }
+$env = @include dirname(__DIR__).'/.env.local.php';
 
+if ('cli' !== PHP_SAPI && is_array($env) && 'prod' === ($_SERVER['APP_ENV'] ?? $_ENV['APP_ENV'] ?? $env['APP_ENV'] ?? null)) {
+    $_SERVER += $env;
+    $_ENV += $env;
+} elseif (!class_exists(Dotenv::class)) {
+    throw new RuntimeException('Please run "composer require symfony/dotenv" to load the ".env" files configuring the application.');
+} else {
     (new Dotenv())->loadEnv(dirname(__DIR__).'/.env');
+
+    if ('prod' === ($_SERVER['APP_ENV'] ?? $_ENV['APP_ENV'] ?? null)) {
+        $env = array_intersect_key($_SERVER, array_flip(explode(',', $_SERVER['SYMFONY_DOTENV_VARS'])));
+        $env = "<?php // This file has been generated by config/bootstrap.php\n\nreturn ".var_export($env, true).";\n";
```

Comment:
> This file was generated by config/bootstrap.php - update your .env files and re-build the cache to update it

Btw, is this accurate? Will re-running `cache:warmup --env=prod` cause this to be overwritten?

Reply:
message updated